cmd/state-remote-installer/main.go

```diff
@@ -0,0 +1,204 @@
+	return &Params{}
+}
+
+func main() {
+	var exitCode int
+
+	var an analytics.Dispatcher
+
+	var cfg *config.Instance
+
+	// Handle things like panics, exit codes and the closing of globals
+	defer func() {
+		if panics.HandlePanics(recover(), debug.Stack()) {
+			exitCode = 1
+		}
+
+		if err := cfg.Close(); err != nil {
+			logging.Error("Failed to close config: %w", err)
+		}
+
+		if err := events.WaitForEvents(5*time.Second, rollbar.Wait, an.Wait, logging.Close); err != nil {
+			logging.Warning("state-remote-installer failed to wait for events: %v", err)
+		}
+		os.Exit(exitCode)
+	}()
+
+	// Set up verbose logging
+	logging.CurrentHandler().SetVerbose(os.Getenv("VERBOSE") != "")
+	// Set up rollbar reporting
+	rollbar.SetupRollbar(constants.StateInstallerRollbarToken)
+
+	// Allow starting the installer via a double click
+	captain.DisableMousetrap()
+
+	// Set up configuration handler
+	cfg, err := config.New()
+	if err != nil {
+		logging.Error("Could not set up configuration handler: " + errs.JoinMessage(err))
+		fmt.Fprintln(os.Stderr, err.Error())
+		exitCode = 1
+	}
+
+	rollbar.SetConfig(cfg)
+
+	// Set up output handler
+	out, err := output.New("plain", &output.Config{
+		OutWriter:   os.Stdout,
+		ErrWriter:   os.Stderr,
+		Colored:     true,
+		Interactive: false,
+	})
+	if err != nil {
+		logging.Error("Could not set up output handler: " + errs.JoinMessage(err))
+		fmt.Fprintln(os.Stderr, err.Error())
+		exitCode = 1
+		return
+	}
+
+	an = sync.New(cfg, nil)
+
+	// Set up prompter
+	prompter := prompt.New(true, an)
+
+	params := newParams()
+	cmd := captain.NewCommand(
+		"state-installer",
+		"",
+		"Installs or updates the State Tool",
+		primer.New(nil, out, nil, nil, nil, nil, cfg, nil, nil, an),
+		[]*captain.Flag{ // The naming of these flags is slightly inconsistent due to backwards compatibility requirements
+			{
+				Name:        "channel",
+				Description: "Defaults to 'release'.  Specify an alternative channel to install from (eg. beta)",
+				Value:       &params.branch,
+			},
+			{
+				Shorthand: "b", // backwards compatibility
+				Hidden:    true,
+				Value:     &params.branch,
+			},
+			{
+				Name:        "version",
+				Shorthand:   "v",
+				Description: "The version of the State Tool to install",
+				Value:       &params.version,
+			},
+			{
+				Name:        "force",
+				Shorthand:   "f",
+				Description: "Force the installation, overwriting any version of the State Tool already installed",
+				Value:       &params.force,
+			},
+		},
+		[]*captain.Argument{},
+		func(ccmd *captain.Command, args []string) error {
+			return execute(out, prompter, cfg, an, args, params)
+		},
+	)
+
+	err = cmd.Execute(os.Args[1:])
+	if err != nil {
+		if locale.IsInputError(err) {
+			logging.Error("Installer input error: " + errs.JoinMessage(err))
+		} else {
+			multilog.Critical("Installer error: " + errs.JoinMessage(err))
+		}
+
+		exitCode = errs.UnwrapExitCode(err)
+		out.Error(err.Error())
```

Comment:
We have a new shared framework for handling errors across applications: https://github.com/ActiveState/cli/blob/9dce4312782cab47fc3b443617d6ed60b6e67a70/pkg/cmdlets/errors/errors.go

Please use it.

Sample usage:
https://github.com/ActiveState/cli/pull/2065/commits/fcc04c3dfc7665b3c3b6906a8f62b4e57b3ee1ba